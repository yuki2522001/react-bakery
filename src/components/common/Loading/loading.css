.loading-container {
  position: fixed;
  display: flex;
  z-index: 20;
  height: 32px;
  width: 32px;
  margin: auto;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.circle {
	animation: grow 1s linear infinite;
	background-color: var(--first);
	border-radius: 50%;
	display: inline-block;
	margin: -5px;
	height: 40px;
	width: 40px;
	transform: scale(0);
}

.circle:nth-of-type(2) {
	animation-delay: .65s;
	background-color: var(--secondary);
}

.circle:nth-of-type(3) {
	animation-delay: .85s;
	background-color: var(--third);
}

@keyframes grow {
	50% {
		transform: scale(0.75);
	}
}