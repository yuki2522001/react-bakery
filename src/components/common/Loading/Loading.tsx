import React from "react";
import "./loading.css";

const Loading: React.FC = () => {
  return (
    <div data-testid="loading-page" className="loading-container">
      <div className="circle"></div>
      <div className="circle"></div>
      <div className="circle"></div>
    </div>
  );
};

export default Loading;
