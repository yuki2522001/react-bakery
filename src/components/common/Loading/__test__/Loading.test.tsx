import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Loading from "../Loading";

describe("Loading component", () => {
  test("should render loading component", () => {
    const { getByTestId } = render(<Loading />);
    expect(getByTestId("loading-page")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });
})