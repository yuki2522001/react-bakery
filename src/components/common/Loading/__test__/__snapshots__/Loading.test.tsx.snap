// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loading-container"
    data-testid="loading-page"
  >
    <div
      class="circle"
    />
    <div
      class="circle"
    />
    <div
      class="circle"
    />
  </div>
</DocumentFragment>
`;
